## This script is used for expression-level dominant (ELD) and transgressive expression analysis in short-liguled T. miscellus (Tms)

In [1]:
import pandas as pd
import os

In [2]:
os.chdir("/ufrc/soltis/shan158538/TragFL_NewAnalysis/OutPut/DE_analysis/Additive_expression_Tms/")

### 1. Transgressive expression analysis

In [3]:
## Input files
nonAdditive = pd.read_table("DESeq2_Tms_nonAdditive_loci.txt", sep = "\t")
DE_Tms_higher_than_Tdu = pd.read_table("DESeq2_DE_loci_Tms_higher_than_Tdu.txt", sep = "\t")
DE_Tms_higher_than_Tpr = pd.read_table("DESeq2_DE_loci_Tms_higher_than_Tpr.txt", sep = "\t")

DE_Tdu_higher_than_Tms = pd.read_table("DESeq2_DE_loci_Tdu_higher_than_Tms.txt", sep = "\t")
DE_Tpr_higher_than_Tms = pd.read_table("DESeq2_DE_loci_Tpr_higher_than_Tms.txt", sep = "\t")

In [4]:
nonAdditive.head()

baseMean    ...         padj
Tpr_TRINITY_DN1451_c0_g1|Tdu_TRINITY_DN16990_c3_g1  100.236819    ...     0.001337
Tpr_TRINITY_DN10853_c1_g4|Tdu_TRINITY_DN21860_c...  750.334395    ...     0.006242
Tpr_TRINITY_DN12798_c2_g2|Tdu_TRINITY_DN24179_c...   58.686028    ...     0.001516
Tpr_TRINITY_DN8521_c0_g1|Tdu_TRINITY_DN16367_c0_g1    4.080691    ...     0.048222
Tpr_TRINITY_DN11678_c3_g2|Tdu_TRINITY_DN24143_c...   57.367802    ...     0.000751

[5 rows x 6 columns]

In [11]:
## Intersections
## Tms higher than both Tdu and Tpr
nonAdditive_Tms_higher_than_Tdu = pd.merge(nonAdditive, 
                                           DE_Tms_higher_than_Tdu, how='inner', 
                                           left_index=True, right_index=True)
nonAdditive_Tms_higher_than_Tdu_and_Tpr = pd.merge(nonAdditive_Tms_higher_than_Tdu, 
                                                   DE_Tms_higher_than_Tpr, 
                                                   how='inner', left_index=True, right_index=True)

In [12]:
nonAdditive_Tms_higher_than_Tdu_and_Tpr.shape

(273, 18)

In [13]:
## Intersections
## Tms lower than both Tdu and Tpr
nonAdditive_Tms_lower_than_Tdu = pd.merge(nonAdditive, 
                                          DE_Tdu_higher_than_Tms, how='inner', 
                                          left_index=True, right_index=True)
nonAdditive_Tms_lower_than_Tdu_and_Tpr = pd.merge(nonAdditive_Tms_lower_than_Tdu, 
                                                  DE_Tpr_higher_than_Tms, 
                                                  how='inner', left_index=True, right_index=True)

In [14]:
nonAdditive_Tms_lower_than_Tdu_and_Tpr.shape

(480, 18)

In [15]:
## Write out
nonAdditive_Tms_higher_than_Tdu_and_Tpr.to_csv("Tms_transgressive_higher.txt", columns=[], header=False)
nonAdditive_Tms_lower_than_Tdu_and_Tpr.to_csv("Tms_transgressive_lower.txt", columns=[], header=False)

**273** loci are transgressively higher expressed in Tms

**480** loci are transgressively lower expressed in Tms

### 2. Expression-level dominant

In [16]:
## Inputfiles
DE_Tdu_higher_than_Tpr = pd.read_table("DESeq2_DE_loci_Tdu_higher.txt", sep = "\t")
DE_Tpr_higher_than_Tdu = pd.read_table("DESeq2_DE_loci_Tpr_higher.txt", sep = "\t")
Tms_same_as_Tdu = pd.read_table("DESeq2_noneDE_loci_Tms_Tdu.txt", sep = "\t")
Tms_same_as_Tpr = pd.read_table("DESeq2_noneDE_loci_Tms_Tpr.txt", sep = "\t")

In [17]:
## Intersections
## Expression-level dominant of Tdu
nonAdditive_ELDofTdu = pd.merge(nonAdditive, Tms_same_as_Tdu, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTdu_TduHigher = pd.merge(nonAdditive_ELDofTdu, DE_Tdu_higher_than_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTdu_TprHigher = pd.merge(nonAdditive_ELDofTdu, DE_Tpr_higher_than_Tdu, how='inner', left_index=True, right_index=True)

In [18]:
nonAdditive_ELDofTdu_TduHigher.shape

(88, 18)

In [19]:
nonAdditive_ELDofTdu_TprHigher.shape

(92, 18)

In [20]:
## Intersections
## Expression-level dominant of Tpr
nonAdditive_ELDofTpr = pd.merge(nonAdditive, Tms_same_as_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTpr_TduHigher = pd.merge(nonAdditive_ELDofTpr, DE_Tdu_higher_than_Tpr, how='inner', left_index=True, right_index=True)
nonAdditive_ELDofTpr_TprHigher = pd.merge(nonAdditive_ELDofTpr, DE_Tpr_higher_than_Tdu, how='inner', left_index=True, right_index=True)

In [21]:
nonAdditive_ELDofTpr_TduHigher.shape

(72, 18)

In [22]:
nonAdditive_ELDofTpr_TprHigher.shape

(85, 18)

In [23]:
## Write out
nonAdditive_ELDofTdu_TduHigher.to_csv("Tms_ELDofTdu_TduHigher.txt", columns=[], header=False)
nonAdditive_ELDofTdu_TprHigher.to_csv("Tms_ELDofTdu_TprHigher.txt", columns=[], header=False)

nonAdditive_ELDofTpr_TduHigher.to_csv("Tms_ELDofTpr_TduHigher.txt", columns=[], header=False)
nonAdditive_ELDofTpr_TprHigher.to_csv("Tms_ELDofTpr_TprHigher.txt", columns=[], header=False)

**88** loci showed ELD of Tdu with higher expression in Tdu than Tpr

**92** loci showed ELD of Tdu with higher expression in Tpr than Tdu

**72** loci showed ELD of Tpr with higher expression in Tdu than Tpr

**85** loci showed ELD of Tpr with higher expression in Tpr than Tdu